app/metal-controller-manager/controllers/server_controller.go

```diff
@@ -129,6 +145,21 @@ func (r *ServerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		}
 	}
 
+	if s.ObjectMeta.DeletionTimestamp.IsZero() {
+		if finalizerIndex(s.ObjectMeta.Finalizers, serverBindingFinalizer) == -1 {
+			s.ObjectMeta.Finalizers = append(s.ObjectMeta.Finalizers, serverBindingFinalizer)
+			if err := r.Update(ctx, &s); err != nil {
```

Comment:
We are using Patch method to avoid conflicts iirc

Reply:
thanks, fixed.